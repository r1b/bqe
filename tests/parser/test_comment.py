import pytest


@pytest.mark.parametrize(
    "sql",
    (
        pytest.param(
            """
            -- myfile.sql
            SELECT 1
            """,
            id="comment_sql_leading",
        ),
        pytest.param(
            """
            SELECT
                1, -- one
                2
            """,
            id="comment_sql_expr",
        ),
        pytest.param(
            """
            SELECT 1
            -- one
            """,
            id="comment_sql_trailing",
        ),
        pytest.param(
            """
            # TODO: SELECT 2
            SELECT 1
            """,
            id="comment_sh_leading",
        ),
        pytest.param(
            """
            SELECT
                1, # one
                2
            """,
            id="comment_sh_expr",
        ),
        pytest.param(
            """
            SELECT 1
            # one
            """,
            id="comment_sh_trailing",
        ),
        pytest.param(
            """
            /* TODO: SELECT 2 */
            SELECT 1
            """,
            id="comment_c_leading",
        ),
        pytest.param(
            """
            SELECT
                1, /* one */
                2
            """,
            id="comment_c_expr",
        ),
        pytest.param(
            """
            SELECT 1
            /* one */
            """,
            id="comment_c_trailing",
        ),
    ),
)
def test_comment_ok(sql, assert_parse_tree):
    assert_parse_tree(sql)
