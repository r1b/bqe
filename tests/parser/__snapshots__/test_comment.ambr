# serializer version: 1
# name: test_comment_ok[comment_c_expr]
  '''
  SELECT
      1, /* one */
      2
  
  query_expr
    select
      select_list
        select_expr
          integer_literal	1
        select_expr
          integer_literal	2
  
  '''
# ---
# name: test_comment_ok[comment_c_leading]
  '''
  /* TODO: SELECT 2 */
  SELECT 1
  
  query_expr
    select
      select_list
        select_expr
          integer_literal	1
  
  '''
# ---
# name: test_comment_ok[comment_c_trailing]
  '''
  SELECT 1
  /* one */
  
  query_expr
    select
      select_list
        select_expr
          integer_literal	1
  
  '''
# ---
# name: test_comment_ok[comment_sh_expr]
  '''
  SELECT
      1, # one
      2
  
  query_expr
    select
      select_list
        select_expr
          integer_literal	1
        select_expr
          integer_literal	2
  
  '''
# ---
# name: test_comment_ok[comment_sh_leading]
  '''
  # TODO: SELECT 2
  SELECT 1
  
  query_expr
    select
      select_list
        select_expr
          integer_literal	1
  
  '''
# ---
# name: test_comment_ok[comment_sh_trailing]
  '''
  SELECT 1
  # one
  
  query_expr
    select
      select_list
        select_expr
          integer_literal	1
  
  '''
# ---
# name: test_comment_ok[comment_sql_expr]
  '''
  SELECT
      1, -- one
      2
  
  query_expr
    select
      select_list
        select_expr
          integer_literal	1
        select_expr
          integer_literal	2
  
  '''
# ---
# name: test_comment_ok[comment_sql_leading]
  '''
  -- myfile.sql
  SELECT 1
  
  query_expr
    select
      select_list
        select_expr
          integer_literal	1
  
  '''
# ---
# name: test_comment_ok[comment_sql_trailing]
  '''
  SELECT 1
  -- one
  
  query_expr
    select
      select_list
        select_expr
          integer_literal	1
  
  '''
# ---
